# <font color=green>Relatório de Processo Diário -
<font color=cyan> DESENVOLVIMENTO DE SOFTWARE




##<font color= violet>Importando a biblioteca Pandas

In [ ]:
import pandas as pd

###<font color=violet>*Função para criar o dataFrame

In [ ]:
def criar_dataframe():
    #Criaando o DataFrame com os dados fornecidos.
    dados = {
        'Dia da Semana': ['Segunda-feira', 'Terça-feira', 'Quarta-feira', 'Quinta-feira', 'Sexta-feira', 'Sábado', 'Domingo'],
        'Horas Trabalhadas': [6, 7, 8, 6, 7, 5, 4],
        'Bugs Corrigidos': [3, 2, 1, 4, 3, 2, 1],
        'Tarefas Concluídas': [5, 4, 6, 4, 5, 3, 2]
    }
    return pd.DataFrame(dados)

###<font color=violet> Calculando as métricas com base nos dados fornecidos

In [ ]:
def calcular_metricas(dados):
    #Calculando as métricas solicitadas com base nos dados fornecidos.
    total_horas_trabalhadas = dados['Horas Trabalhadas'].sum()
    media_diaria_horas_trabalhadas = dados['Horas Trabalhadas'].mean()
    total_bugs_corrigidos = dados['Bugs Corrigidos'].sum()
    media_diaria_bugs_corrigidos = dados['Bugs Corrigidos'].mean()
    total_tarefas_concluidas = dados['Tarefas Concluídas'].sum()
    media_diaria_tarefas_concluidas = dados['Tarefas Concluídas'].mean()

    dados['Produtividade Diária'] = dados['Tarefas Concluídas'] / dados['Horas Trabalhadas']
    produtividade_diaria = dados['Produtividade Diária'].mean()

    return (
        total_horas_trabalhadas, media_diaria_horas_trabalhadas,
        total_bugs_corrigidos, media_diaria_bugs_corrigidos,
        total_tarefas_concluidas, media_diaria_tarefas_concluidas,
        produtividade_diaria
    )

##<font color=violet>Criando a Função pra Gerar o Relatório de Dados

In [ ]:
def gerar_relatorio(dados):
    #Gerando o relatório de progresso diário com base nos dados fornecidos.
    total_horas_trabalhadas, media_diaria_horas_trabalhadas, \
    total_bugs_corrigidos, media_diaria_bugs_corrigidos, \
    total_tarefas_concluidas, media_diaria_tarefas_concluidas, \
    produtividade_diaria = calcular_metricas(dados)

     # Montando o relatório
    relatorio = f"RELATÓRIO DE PROGRESSO DIÁRIO - DESENVOLVIMENTO DE SOFTWARE\n"
    relatorio += f"Período: {dados['Dia da Semana'].iloc[0]} a {dados['Dia da Semana'].iloc[-1]}\n\n"
    relatorio += f"1. Total de Horas Trabalhadas: {total_horas_trabalhadas} horas\n"
    relatorio += f"2. Média Diária de Horas Trabalhadas: {media_diaria_horas_trabalhadas:.2f} horas\n"
    relatorio += f"3. Total de Bugs Corrigidos: {total_bugs_corrigidos} bugs\n"
    relatorio += f"4. Média Diária de Bugs Corrigidos: {media_diaria_bugs_corrigidos:.2f} bugs por dia\n"
    relatorio += f"5. Total de Tarefas Concluídas: {total_tarefas_concluidas} tarefas\n"
    relatorio += f"6. Média Diária de Tarefas Concluídas: {media_diaria_tarefas_concluidas:.2f} tarefas por dia\n"
    relatorio += f"7. Produtividade Diária (Tarefas Concluídas por Hora): {produtividade_diaria:.2f} tarefas por hora de trabalho"

    return relatorio

In [ ]:

def main():
    # Criando o DataFrame com os dados fornecidos
    dados = criar_dataframe()

    # Gerando o relatório
    relatorio = gerar_relatorio(dados)

    # Imprimindo o relatório
    print(relatorio)

# Executando o código principal
main()

RELATÓRIO DE PROGRESSO DIÁRIO - DESENVOLVIMENTO DE SOFTWARE
Período: Segunda-feira a Domingo

1. Total de Horas Trabalhadas: 43 horas
2. Média Diária de Horas Trabalhadas: 6.14 horas
3. Total de Bugs Corrigidos: 16 bugs
4. Média Diária de Bugs Corrigidos: 2.29 bugs por dia
5. Total de Tarefas Concluídas: 29 tarefas
6. Média Diária de Tarefas Concluídas: 4.14 tarefas por dia
7. Produtividade Diária (Tarefas Concluídas por Hora): 0.66 tarefas por hora de trabalho


Com base nos resultados obtidos, a equipe de desenvolvimento trabalhou em média cerca de 6.14 horas por dia, verificou em média 2.29 bugs e concluiu cerca de 4.14 tarefas diariamente. A produtividade geral da equipe foi baixa, com apenas 0.10 tarefas concluídas por hora de trabalho.





## Insights e Recomendações

Com base nos dados analisados, podemos fazer algumas observações:

1. A quantidade média de horas trabalhadas por dia está dentro de uma faixa razoável e equilibrada, sugerindo que os funcionários não estão sobrecarregados ou ociosos.

2. A média diária de bugs corrigidos está relativamente baixa. Isso pode indicar a necessidade de revisar os processos de verificação de código e aprofundar as práticas de desenvolvimento para melhorar a qualidade do software.

3. A produtividade diária indica que os funcionários estão concluindo, em média, menos de uma tarefa por hora trabalhada. Isso pode ser um indicativo de que as tarefas estão muito complexas ou mal dimensionadas, o que pode ser revisto para melhorar a eficiência da equipe.

Com base nas observações acima, recomenda-se:

- Realizar uma análise mais detalhada dos bugs identificados para entender suas causas e tomar medidas preventivas para reduzi-los.

- Investir em treinamentos e capacita